### DISCUSSION TITLE
CORS ERROR

### DISCUSSION THREAD

**Discussion by aly8i** [vercel/community#1009] [Help] [answered]
I have a Nextjs application deployed with Vercel fetching an authentication response from an express server deployed on heruko.

Please provide a solution.

# here is the error:
> Access to fetch at 'https://restoserver.herokuapp.com/auth/login/success' from origin 'https://resturantz.vercel.app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

# here is the CORS in express server:
``app.use( cors({ origin: `${process.env.CLIENT_URL}`, methods: "GET,POST,PUT,DELETE", credentials: true, }) );``

# here is the fetch request on Next Js:
``fetch(`${process.env.NEXT_PUBLIC_SERVER_URL}/auth/login/success`, { method: "GET", credentials: "include", headers: { Accept: "application/json", "Content-Type": "application/json", "Access-Control-Allow-Credentials": true, "Access-Control-Allow-Origin": `${process.env.NEXT_PUBLIC_SERVER_URL}` } }) .then((response) => { //more code }) //more code``

In my development environment, communication between localhost:3000 (the website) and localhost:5000 (the server), worked fine. while in production, the server isn't on the same url as the website. The communication failed.

**Comment by aly8i** [comment]
@amyegan check this

**Comment by amyegan** [comment, marked as answer]
Based on the error message, it looks like you need to adjust the CORS settings on your Heroku Express server. You may need to use [Express CORS Middleware](https://expressjs.com/en/resources/middleware/cors.html) or try something like this Stack Overflow answer - https://stackoverflow.com/a/40026625
